Doc/library/ctypes.rst

```diff
@@ -1025,6 +1025,23 @@ As we can easily check, our array is sorted now::
    1 5 7 33 99
    >>>
 
+The function factories can be used as decorator factories, so we may as well
+write::
+
+   >>> @CFUNCTYPE(c_int, POINTER(c_int), POINTER(c_int))
+   ... def py_cmp_func(*args):
```

Comment:
Why didn't you reuse the `py_cmp_func` function in line 1008? IMO, the following snippet should work:

```py
@CFUNCTYPE(c_int, POINTER(c_int), POINTER(c_int))
def py_cmp_func(a, b):
    print("py_cmp_func", a[0], b[0])
    return a[0] - b[0]
```

There is even a test for that same function:

https://github.com/python/cpython/blob/b91a3a0d61596cafb1b46c98fab65fee16a8bbbb/Lib/ctypes/test/test_callbacks.py#L186-L188

Reply:
I followed the proposal from David W. Lambert in the issue, but you are right, I tested the decorator with the function from line 1008 and works as well.